### PR TITLE
`Exam mode`: Fix students table resize functionality

### PR DESCRIPTION
**Pull request by beyzaaltuntas**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Fixes https://github.com/ls1intum/Artemis/issues/7915
Probably fixes https://github.com/ls1intum/Artemis/issues/7573 also
### Description

While resizing the page, the students table on the exam page failed to adjust accordingly. The resizing behavior of the students table has been improved.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students

1. Be signed in as an instructor
2. Select a course
3. Click on manage
4. Select the exams tab
5. Find an exam and click on the students button
6. Resize the page

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### Screenshots

https://github.com/ls1intum/Artemis/assets/30345525/21b2d959-da2b-45cc-bc07-74b5a4279a7c

## Summary by CodeRabbit

- **Style Updates**
	- Added styles for overflow, width, and flex properties in the `exam-students.component.scss` file.

- **New Features**
	- Implemented flexible column width mode for enhanced data presentation in tables.

- **Bug Fixes**
	- Standardized row heights across tables for consistency.

### CODE REVIEW

**Review by LenaKahle — CHANGES_REQUESTED** [review]
When narrowing the size of the window, all rows get an individual scroll bar, is that intended behavior?
image

**Review by konrad2002 — CHANGES_REQUESTED** [review]
on mac os the problem mentioned by @LenaKahle is not as bad looking as on Windows but the single scrollbars also cover parts of the buttons when scrolling. Manually tested on legacy ts1.

grafik

**Comment by beyzaaltuntas** [comment]
@LenaKahle @konrad2002 I removed the scrollbar, could you please check again?

**Review by MaximilianJG — APPROVED** [review]
Table is adjusting correctly. Everything works as expected. Screenshot 2024-01-24 at 09 22 15

**Review by nata-81 — DISMISSED** [review]
Tested on ts4 works as expected.

**Review by Gusti2010 — APPROVED** [review]
manually tested on ts2_Legacy. Works as expected!

**Review by konrad2002 — APPROVED** [review]
re-approve: manually tested on legacy ts11, works as expected

**Review by LenaKahle — DISMISSED** [review]
Approve, tested on ts1.

**Review by AntonGeTUM — DISMISSED** [review]
Tested on ts1, works as demonstrated

**Review by nata-81 — APPROVED** [review]
Tested on ts2 works as expected.

**Review by MaximilianAnzinger — CHANGES_REQUESTED** [review]

**Review by MaximilianJG — DISMISSED** [review]

File: src/main/webapp/app/exam/manage/students/exam-students.component.scss [hunk 1]
```diff
@@ -58,5 +58,26 @@ jhi-exam-students {
             animation-duration: map-get($vars, animation-duration);
             animation-timing-function: ease-out;
         }
+        overflow-x: auto;
+        overflow-y: hidden;
+    }
+
+    .datatable-header-inner {
+        width: 100% !important;
```
[the remark this answers is not in the document]
Comment (MaximilianAnzinger) [reply src/main/webapp/app/exam/manage/students/exam-students.component.scss:69]: Yes we use Bootstrap

**Review by Gusti2010 — APPROVED** [review]
Reapprove. Manually tested on ts11_Legacy.

**Review by LenaKahle — APPROVED** [review]
Reapprove.

**Review by AntonGeTUM — APPROVED** [review]
reapprove

**Review by MaximilianJG — APPROVED** [review]

**Pull request merged**